### ISSUE TITLE
HTTP server doesn't handle cookies containing '='

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10703]
|<img alt="abram's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| abram reported|
|-|-|
|Trac ID|trac#1051|
|Type|defect|
|Created|2005-05-25 04:19:14Z|

Attachments:

* [http_1051.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/929/929f9640bc8269918a6209919a8e2cc25ec65bf6/f1fbb2b8a33b5c0bd62770c59f2748c0e92dc745.diff) (1785 bytes) - added by therve on 2006-10-04 08:43:29Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1051 1051
type__defect defect
reporter__abram abram
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1116994754000000 1116994754000000
changetime__1159952844000000 1159952844000000
version__ 
owner__ 
cc__Tv cc__jknight cc__abram
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="abram's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|abram commented|
|-|-|

```
#!html
<pre>
When a cookie contains the value '=', the Twisted HTTP server fails to parse it.
This bug was reported once before, in issue 190
(http://twistedmatrix.com/bugs/#190), but it was never really fixed, just
wrapped in a try/except block so it wouldn't raise an exception. The correct fix
is to change line 537 of twisted/web/http.py from:
k, v = cook.split('=')

to:
k, v = cook.split('=', 1)
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

Attached a patch with the obvious fix, with a testcase.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml set owner to @therve|
|-|-|

therve, you are a legend. please add a docstring for testCookies and then re-assign to me. I'll merge as soon as I may.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @jml|
|-|-|

Wouhou, I'm blushing.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set status to `closed`|
|-|-|

(In [18383]) Allow '=' to be in the value of a cookie.

 * Author: therve
 * Reviewer: jml
 * Fixes [#10703](https://github.com/twisted/twisted/issues/10703)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
